manifests/config.pp

```diff
@@ -59,6 +59,38 @@
 
     }
 
+    'Suse': {
+      $pamconfig_mkhomedir_check_cmd  = '/usr/sbin/pam-config -q --mkhomedir | grep session:'
+      $pamconfig_check_cmd            = '/usr/sbin/pam-config -q --sss | grep session:'
+      $nsconfig_check_cmd             = '/usr/bin/test $( /bin/grep -cE "(passwd|group|sudoers).*sss" /etc/nsswitch.conf ) -gt 2'
+      $nsconfig_cmd                   = '/bin/sed -i -e \'/^\(passwd\|group\|sudoers\):/{/sss\b/ b;s/^\(.*\)$/\1 sss/}\' /etc/nsswitch.conf'
+
+      if $mkhomedir {
+
+        exec { 'pam-config -a --mkhomedir':
```

Comment:
This file is getting rather large with execs...
It might be nice to have a folder called config with a suse.pp, redhat.pp, debian.pp etc in it.
Then just include the appropriate class, for the correct execs...

Just a thought...